### PR TITLE
Adds Research cyborgs [Modularish] (oh no)

### PR DESCRIPTION
**Pull request by pixelkitty286**
## About The Pull Request

This PR added a new module for cyborgs it also ~~replaces the malfunctioning AI cyborg factory~~ with the ability when hacked to manufacture cyborgs manually (Cyborg building CO-OP) Malfunctioning AI cyborg factory was added back.

The PR also adds a research cyborg exclusive jaunting module, much like the ninja energy sword and hierophant, when they are emagged or hacked.

An alterative way was added to remove cells on cyborgs to allow for cyborgs to upgrade ONLY cells. Other upgrades are restricted to balance them.

This comes with a few base sprites for the research cyborgs more will be added later or in a future PR.

## Why It's Good For The Game

I've always wondered why the research department never had cyborgs of their own, but own them.

Research cyborgs are for a dedicated cyborg type assisting the research department.

~~Disabling the robot factory was done because the printer would just shit 3 to 4 cyborgs at once and make them out of nothing.~~

It was added with and optional Cyborg CO-OP to encourage the cyborgs and the AI to make cyborgs them self if they wish for more cyborgs.

Research cyborgs were given jaunting and a stun arm to hopefully allow them to evade capture. In other words they are to run when hunted as they are mostly defenseless (fighting back is a bad idea.)

## Proof Of Testing

NOTE: These images taken were during development and may be missing a few items. Also some items may have been removed wile testing.

LIST OF ITEMS REMOVED:
Dropper
Syringe
Mech gripper

Stock
dreamseeker_zPm0yyoOZC

Fully upgraded
dreamseeker_qyG5Oq1cav

Emagged
dreamseeker_nQcm41bycd
Note: NO EMAG
dreamseeker_tNIapvYTHM

## Changelog

:cl:
add: Research cyborgs
add: ability to pry cyborg cells out 
del: ~~disabled malf cyborg factory~~
qol: made an alt method removing cells
balance: rebalanced cyborg making
image: Sprites for research cyborgs
code: what ever needed to get the cyborgs working
/:cl:

### CODE REVIEW

**Comment by pixelkitty286** [comment]
upstream merge has caused new bugs and will have to put this on hold

**Comment by pixelkitty286** [comment]
oh uh, it's ready

**Comment by StrangeWeirdKitten** [comment]
qol: made an alt method removing cells
balance: rebalanced cyborg making
image: Sprites for research cyborgs
code: what ever needed to get the cyborgs working

Don't entirely need this in the changelog.

**Comment by pixelkitty286** [comment]
> qol: made an alt method removing cells balance: rebalanced cyborg making image: Sprites for research cyborgs code: what ever needed to get the cyborgs working
> 
> Don't entirely need this in the changelog.

I'm just used to doing that since I was required to where I last made PRs at.

**Comment by pixelkitty286** [comment]
Got a round in to test my stuff.  The module mostly works only problem ran into seems to being unable to pick up rods or eject some stuff, mainly cells, I will be adding rods as a pick up and synth repairs are kinda scuffed, but I will be looking more into it the more I test in real rounds (I can not add any more items kinda at the limit) Other than those two issues seems my local tests all worked as intended.
dreamseeker_yvQzKjWJVT

**Comment by StrangeWeirdKitten** [comment]
Why does it need the needle and the dropper?

**Comment by StrangeWeirdKitten** [comment]
Also synths have this issue where some of the surgeries*might* require medical tools for surgery. But they should be capable of basic repair and revival

**Comment by ChaosCzar** [comment]
> Why does it need the needle and the dropper?

For assisting with Xenobio duties, I'd assume.

**Comment by pixelkitty286** [comment]
> Also synths have this issue where some of the surgeries_might_ require medical tools for surgery. But they should be capable of basic repair and revival

I can add the tools required but do know if i do it the inventory will take half of the screen and also would pretty much make Medi borgs useless

**Comment by pixelkitty286** [comment]
> > Why does it need the needle and the dropper?
> 
> For assisting with Xenobio duties, I'd assume.

yes i was told to do such I will be taking suggestions to what items i should remove to cut bloat down

**Comment by pixelkitty286** [comment]
Added rods to be picked up by the sheet apparatus.

**Comment by pixelkitty286** [comment]
Stock RPED is broken this seems to be an issue upstream.

**Comment by YakumoChen** [comment]
Initial thoughts is that malf changes should be atomised

The borg model itself also seems like fairly ludicrous power creep, being able to build more borgs and mechs and take out the robotics middle man is bonkers

**Comment by pixelkitty286** [comment]
> Initial thoughts is that malf changes should be atomised
> 
> The borg model itself also seems like fairly ludicrous power creep, being able to build more borgs and mechs and take out the robotics middle man is bonkers
> 
Mech building is allowed but you can not move them so good luck. Mod building is allowed but is very slow.

Borg building is locked be hind illegal tech and emag.
> It's almost all the best parts of engi and medborg plus a ton more
> 
Jessie what are you talking about
> Also "it's defenseless when emagged" but bro you can teleport and stun, that's the most powerful combo outside of "literally a gun"

The stun arm is really useless it has a cool off of 4 seconds.

Jaunting is not able to go though walls and has a penalty on using it 3200 KJ of power along with a reasonable cool down for recharge.

**Comment by StrangeWeirdKitten** [comment]
> Initial thoughts is that malf changes should be atomised
No.

**Comment by R3dtail** [comment]
As a gut reaction it doesn't feel really great that they can do most of robotics outside of moving the mechs.  Roboticist is already a very limited job that sometimes feels crowded with two people.  Sure, there's 'butting in and telling them to buzz off' but that's not fun for anyone to do.  It doesn't feel good to be replaced, borgs should be able to help, sure.  But being replaced doesn't feel great.

**Comment by pixelkitty286** [comment]
Added cyborg printing back since wow is it painful. Cyborg making is now optional.

**Comment by Eastwardrope** [comment]
Borgs having that power is cool in theory but in practice it ends up being way way too powerful. One of the main weaknesses of borgs is requiring them to have a human to do basic maintence. I think the battery swap thing should take a long period of time to do(something like 10 seconds) as removing their batteries is a big way to gimp borgs. I do think it fits the servers style and vibe but I fear it might be abused hard by Malfs or anyone with a Law Board and a dream.

**Comment by Swiftfeather** [comment]
Idk what the other maintainers think but I don't like this at all.

**Review by Majkl-J — COMMENTED** [review]
Would you consider this complete?

**Comment by pixelkitty286** [comment]
fuck oops. also yes it's ready.

File: code/game/objects/items/robot/items/storage.dm
Comment (Majkl-J) [inline code/game/objects/items/robot/items/storage.dm:267]: Can be made modular with
```
/obj/item/borg/apparatus/sheet_manipulator/Initialize()
	. = ..()
	storable += /obj/item/stack/rods
```

File: code/game/objects/items/robot/robot_upgrades.dm
Comment (Majkl-J) [inline code/game/objects/items/robot/robot_upgrades.dm:573]: Modularly override with

```
/obj/item/borg/upgrade/processor
	name = "medical and research cyborg surgical processor" // BUBBER EDIT Research borgs
	desc = "An upgrade to the Medical and Research model, installing a processor \
		capable of scanning surgery disks and carrying \
		out procedures"

/obj/item/borg/upgrade/processor/Initialize()
	. = ..()
	model_type += /obj/item/robot_model/sci
	model_flags += BORG_MODEL_RESEARCH

File: modular_zubbers/modules/borgs/code/robot_items.dm
Comment (Majkl-J) [inline modular_zubbers/modules/borgs/code/robot_items.dm:118]:
```suggestion
	var/charge_cost = (STANDARD_CELL_CHARGE * 0.0032)
```

File: modular_zubbers/modules/borgs/code/robot_upgrade.dm
Comment (Majkl-J) [inline modular_zubbers/modules/borgs/code/robot_upgrade.dm:36]:
```suggestion
/obj/item/borg/upgrade/healthanalyzer
	name = "Research cyborg advanced Health Analyzer"
	desc = "An upgrade to the Research model cyborg's standard health analyzer."
	icon_state = "cyborg_upgrade3"
	require_model = TRUE
	model_type = list(/obj/item/robot_model/sci)
	model_flags = BORG_MODEL_RESEARCH

	items_to_add = list(/obj/item/healthanalyzer/advanced)
	items_to_remove = list(/obj/item/healthanalyzer)
```

File: modular_zubbers/modules/borgs/code/robot_upgrade.dm
Comment (Majkl-J) [inline modular_zubbers/modules/borgs/code/robot_upgrade.dm:46]:
```suggestion
```

File: modular_zubbers/modules/borgs/code/robot_upgrade.dm
Comment (Majkl-J) [inline modular_zubbers/modules/borgs/code/robot_upgrade.dm:72]:
```suggestion
/obj/item/borg/upgrade/inducer_sci
	name = "Research integrated power inducer"
	desc = "An integrated inducer that can charge a device's internal cell from power provided by the cyborg."
	require_model = TRUE
	model_type = list(/obj/item/robot_model/sci)
	model_flags = BORG_MODEL_RESEARCH

	items_to_add = list(/obj/item/inducer/cyborg/sci)
```

File: modular_zubbers/modules/borgs/code/robot_upgrade.dm
Comment (Majkl-J) [inline modular_zubbers/modules/borgs/code/robot_upgrade.dm:100]:
```suggestion
/obj/item/borg/upgrade/brped
	name = "Research cyborg Rapid Part Exchange Device Upgrade"
	desc = "An upgrade to the Research model cyborg's standard RPED."
	icon_state = "cyborg_upgrade3"
	require_model = TRUE
	model_type = list(/obj/item/robot_model/sci)
	model_flags = BORG_MODEL_RESEARCH

	items_to_add = list(/obj/item/storage/part_replacer/bluespace)
	items_to_remove = list(/obj/item/storage/part_replacer)
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]

**Comment by pixelkitty286** [comment]
Waiting on https://github.com/Bubberstation/Bubberstation/pull/1609 to do the changes.

**Review by Majkl-J — APPROVED** [review]

**Pull request merged**
